docs/guide/custom-ui.md

```diff
@@ -0,0 +1,207 @@
+
+Here is an example of how to do that:
+
+<AposCodeBlock>
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    components: {
+      managerModal: 'AnnouncementManager'
+    }
+  }
+}
+```
+  <template v-slot:caption>
+    modules/announcement/index.js
+  </template>
+</AposCodeBlock>
+
+
+With this configuration, Apostrophe will look for a Vue component called `AnnouncementManager` when the user selects "Announcements" from the admin bar, bypassing `AposDocManager`.
+
+As for the actual Vue code, we would place that in `modules/announcement/ui/apos/components/AnnouncementManager.vue`.
+
+Of course there are other components that can be overridden in this way, and the list is growing over time. Here are the components that can currently be overridden through configuration:
+
+| Module                     | Option             | Default          |
+| -------------------------- | ------------------ | ---------------- |
+| `@apostrophecms/piece-type`  | `components.managerModal` | `AposDocsManager`  |
+| `@apostrophecms/piece-type`  | `components.editorModal`  | `AposDocEditor`    |
+| `@apostrophecms/page`        | `components.managerModal` | `AposPagesManager` |
+| `@apostrophecms/page`        | `components.editorModal`  | `AposDocEditor`    |
+| `@apostrophecms/widget-type` | `components.widgetEditor` | `AposWidgetEditor` |
+| `@apostrophecms/widget-type` | `components.widget`       | `AposWidget`       |
+
+For readability's sake, a `.` is used in the table above to separate sub-properties of `options` (see the example above for what the actual configuration looks like). If an option exists for `@apostrophecms/piece-type` it can be used for any module that extends it.
+
+::: note
+If an option ends in `Modal`, our component is expected to embed the `AposModal` component so that the admin UI can await it with `apos.modal.execute`. For examples, look at the source code of the default components listed above.
+
+The `AposWidgetEditor` component already provides a modal dialog box in which to edit the schema of any widget, so we won't need to configure a replacement unless we want to support editing directly on the page. `AposRichTextWidgetEditor` is an example of how to do this.
+
+The `AposWidget` component has **nothing to do with a typical site visitor experience.** It is used only when displaying our widget while the page is in edit mode. While overriding this component is rare, but the `@apostrophecms/rich-text-widget` module does so to provide a "click the text to edit" experience for rich text widgets. If you're just trying to enhance your widget with frontend JavaScript, you should write a [widget player](custom-widgets.md#client-side-javascript-for-widgets) instead.
+:::
+
+## Registering custom field types
```

Comment:
This should be a code recipe page instead. It's a separate section in the navigation. If you break this out to a new page I can help with the nav.

I see the value for this topic in terms of expanding UI options. It would make sense to have a paragraph that introduces this idea and links over to the code recipe. There's simply more to custom field types than customizing UI.